soh/soh/Enhancements/ResetGameplayFrames.cpp

```diff
@@ -0,0 +1,30 @@
+#include "soh/Enhancements/game-interactor/GameInteractor_Hooks.h"
+#include "soh/ShipInit.hpp"
+
+extern "C" {
+#include "functions.h"
+#include "variables.h"
+}
+
+void ResetGameplayFramesOnSoftReset() {
+    PlayState* play = (PlayState*)gGameState;
+
+    if (gGameState->init == TitleSetup_Init) {
+        play->gameplayFrames = 0;
+    }
+}
+
+void ResetGameplayFramesOnTitleScreenExit() {
```

Comment:
I know this is just copying what we had before, but I figure it's worth pointing out that this is not accurate to what it is actually happening: which is that this is resetting the frame count on any play destroy with the debug save loaded (title screen or LRZ).

I think the intent for the frame reset is to happen when ever we transition from a non-PlayState game state to a PlayState game state.

Not something we need to do in this PR, but should at least track an issue for investigation (e.g. test with hardware)

Reply:
yeah, i didn't want to go down the rabbit hole of that (and autosave stuff) when moving it over (it originally came from https://github.com/HarbourMasters/Shipwright/pull/1792, and there's more `gameplayFrames` stuff in https://github.com/HarbourMasters/Shipwright/pull/3389 too)

Reply:
made an issue https://github.com/HarbourMasters/Shipwright/issues/4985